tools/federation/src/test/java/org/eclipse/rdf4j/federated/FedXBaseTest.java

```diff
@@ -99,6 +101,10 @@ protected void execute(RepositoryConnection conn, String queryFile, String expec
 			}
 
 		} else if (query instanceof GraphQuery) {
+			// Some query results will automatically close themselves when they are exhausted. To properly test that
+			// query results are closed correctly we need to evaluate the query without retrieving any elements.
+			((GraphQuery) query).evaluate().close();
+
```

Comment:
This is one of the two main changes to the tests that make the unclosed iterations visible.